app/src/main/java/com/example/smartshopper/MainActivity.java

```diff
@@ -42,20 +46,9 @@ protected void onCreate(Bundle savedInstanceState) {
         // to make the Navigation drawer icon always appear on the action bar
         Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
 
-        // Recycler View setup
-        deals = new ArrayList<>(); // initiate an empty stickers array (begins with empty recycler view)
-        // Temporary placeholder to display cards in recyclerView
-        User tempuser = new User("megaDeals100");
-        for (int i=0; i < 10; i++) {
-            String dealNumber = String.valueOf(i);
-            Deal tempDeal = new Deal("Deal"+ dealNumber , (double) (i+10), (double) (i+3),dealNumber + "description", "Target", tempuser);
-            deals.add(tempDeal);
-        }
+        // Get deals from firebase:
+        platformHelpers.updateMainRecyclerView(this);
 
-        adapter = new DealAdapter(deals, this);
-        rv_dealsRecyclerView = findViewById(R.id.rv_dealsRecyclerView);
-        rv_dealsRecyclerView.setLayoutManager(new LinearLayoutManager(this));
-        rv_dealsRecyclerView.setAdapter(adapter);
```

Comment:
personal opinion - it feels natural for me to instantiate the RV code block in the _MainActivity.java_ -- if we want to abstract it to it's own utility class -- that is okay -- but i dont think it should be coupled with _updateMainRecyclerView_

```
adapter = new DealAdapter(deals, this);
rv_dealsRecyclerView = findViewById(R.id.rv_dealsRecyclerView);
rv_dealsRecyclerView.setLayoutManager(new LinearLayoutManager(this));
rv_dealsRecyclerView.setAdapter(adapter);
```

Reply:
i think you should refactor it in your own branch.
im not understanding quite what you want.